xdsl/dialects/onnx.py

```diff
@@ -673,6 +675,94 @@ def verify_(self) -> None:
                 f"Only one value attribute must be provided, but {used_attrs} were specified"
             )
 
+    def get_value(self) -> tuple[Any | None, str]:
+        if self.value_float:
+            return self.value_float, "value_float"
+        elif self.value_int:
+            return self.value_int, "value_int"
+        elif self.value_ints:
+            # In the ONNX-MLIR format, integer types are abstracted, so only the integer data is extracted
+            # for printing purposes.
+            arr: list[int] = [value.value.data for value in self.value_ints]
+            return arr, "value_ints"
+        else:
+            # TODO: value_string, value_strings, value_floats
+            raise NotImplementedError()
+
+    def print(self, printer: Printer):
+        if self.value is not None:
+            printer.print_string(" ")
+            printer.print_attribute(self.value)
+        printer.print_string(" ")
+        printer.print_string("{")
+        try:
+            value, value_type = self.get_value()
+        except NotImplementedError:
+            # handle the case where get_value() is not implemented
+            value, value_type = None, None
+        if value and value_type:
+            if value_type == "value_ints":
+                printer.print_string(value_type)
+                printer.print_string(" = ")
+                printer.print(value)
+            else:
+                printer.print_string(value_type)
+                printer.print_string(" = ")
+                printer.print_attribute(value)
+        printer.print_string("}")
```

Comment:
I would recommend changing this to something like this:
```
if self.value is not None:
  printer.print(self.value)
printer.print_attribute_dict(self.attributes, exclude=("value",))
```

Reply:
if i change it to this, this is the issue:
`"value_ints" = [1 : i64, 2 : i64, 3 : i64]} `
but onnx-mlir:
`{value_ints = [1, 2, 3]}`
so we would have to alter the printer

Reply:
Yeah, that's my suggestion

Reply:
so maybe add a special case for `else if self.value_ints is not None`

Reply:
ive just realised that `print_attr_dict` does not have an exclude keyword argument, i think i have to do it manually

Reply:
`Printer.print_op_attributes(reserver_attr_names=...)`

Reply:
> `Printer.print_op_attributes(reserver_attr_names=...)`

just to be clear, you mean you want to change the `Printer` Class not the method that is bounded in the constant class, this is because the attribute name type is printed with strings aswell, so this doesn't work as nicely

Reply:
because if thats the case, i don't see a use for my helper as of such